### PR TITLE
Gm 50 maintainability improvements to components

### PR DESCRIPTION
**Pull request by Gregam3**
Discussed changes implemented

### CODE REVIEW

File: backend/src/main/java/asegroup1/api/services/ServiceImpl.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package asegroup1.api.services;
 
-import asegroup1.api.daos.Dao;
+//import asegroup1.api.daos.Dao;
```
Comment (jamesfernando94) [inline backend/src/main/java/asegroup1/api/services/ServiceImpl.java]: Commented out code
```suggestion
```

**Comment by jamesfernando94** [comment]
Sorry @Gregam3 conflicting files

File: backend/src/main/java/asegroup1/api/daos/DaoImpl.java [hunk 2]
```diff
@@ -27,30 +27,25 @@ protected void setCurrentClass(Class<T> currentClass) {
     @SuppressWarnings("unchecked")
```
Comment (richousrick) [inline backend/src/main/java/asegroup1/api/daos/DaoImpl.java]: This is not needed as the generic type has been specified for class.

File: backend/src/main/java/asegroup1/api/daos/DaoImpl.java [hunk 3]
```diff
@@ -27,30 +27,25 @@ protected void setCurrentClass(Class<T> currentClass) {
...
 
+    public void delete(T t) {
+        entityManager.remove(t);
+    }
+
     public void update(T t) {
         entityManager.merge(t);
     }
 
     @SuppressWarnings("unchecked")
```
Comment (richousrick) [inline backend/src/main/java/asegroup1/api/daos/DaoImpl.java]: This is not needed as the generic type has been specified for class.

File: backend/src/main/java/asegroup1/api/services/user/UserServiceImpl.java [hunk 4]
```diff
@@ -9,10 +9,10 @@
 @Service
 public class UserServiceImpl extends ServiceImpl<UserData> {
 
-    private Dao<UserData> userDataDao;
+    private DaoImpl<UserData> userDataDao;
```
Comment (richousrick) [inline backend/src/main/java/asegroup1/api/services/user/UserServiceImpl.java]: Is this needed, as it is private, and unused in the class

**Pull request closed without merging**